lib/galaxy/jobs/runners/__init__.py

```diff
@@ -623,13 +623,14 @@ def finish_job(self, job_state):
                     stdout = ''
                     stderr = 'Job output not returned from cluster'
                     log.error('(%s/%s) %s: %s' % (galaxy_id_tag, external_job_id, stderr, str(e)))
+                    exit_code_str = "123"  # Indicate that job probaly failed, given that we couldn't collect the Job output
                 else:
                     time.sleep(1)
                 which_try += 1
 
         try:
             # This should be an 8-bit exit code, but read ahead anyway:
-            exit_code_str = open(job_state.exit_code_file, "r").read(32)
+            exit_code_str = open(job_state.exit_code_file, "r").read(32) if not exit_code_str else exit_code_str
```

Comment:
This works properly only if `exit_code_str` is always defined, otherwise this would result in:
```
NameError: name 'exit_code_str' is not defined
```
which is caught by the next `except`, ending up almost always with a value of `"0"`.

Reply:
uff, of course. thanks for the catch.